nuxeo-platform-video-core/src/main/java/org/nuxeo/ecm/platform/video/tools/VideoSlicer.java

```diff
@@ -0,0 +1,151 @@
+            File outputFile;
+            try {
+                outputFile = Framework.createTempFile(FilenameUtils.removeExtension(finalFilename),
+                        "sliced." + FilenameUtils.getExtension(finalFilename));
+            } catch (IOException e) {
+                throw new NuxeoException("VideoSlicer could not set up temporary output file.");
+            }
+            Framework.trackFile(outputFile, this);
+            cmdParameters.put(OUTPUT_FILE_PATH_PARAM, outputFile.getAbsolutePath());
+
+            // check if the slicer should use the duration
+            if (useDuration) {
+                cmdParameters.put(DURATION_PARAM, duration);
+                commandLineName = encode? COMMAND_SLICER_DEFAULT : COMMAND_SLICER_BY_COPY;
+            } else {
+                commandLineName = COMMAND_SLICER_START_AT;
+            }
+        } else if (useDuration) {
+            Path segmentsDirectory;
+            try {
+                segmentsDirectory = Framework.createTempDirectory("Segments");
+            } catch (IOException e) {
+                throw new NuxeoException("VideoSlicer could not create temporary directory for video segments");
+            }
+            File segments = segmentsDirectory.toFile();
+            File dir = new File(segments.getAbsolutePath(), addSuffixToFileName(video.getFilename(), "-%03d"));
+            String outFilePattern = dir.getAbsolutePath();
+
+            cmdParameters.put(OUTPUT_FILE_PATH_PARAM, outFilePattern);
+            cmdParameters.put(DURATION_PARAM, duration);
+            cmdParameters.put(SEGMENTS_PATH, segments.getAbsolutePath());
+            commandLineName = COMMAND_SLICER_SEGMENTS;
+        }
+        return cmdParameters;
+    }
+
+    @Override
+    public BlobHolder buildResult(BlobHolder input, Map<String, String> cmdParams) {
+        if (commandLineName.equals(COMMAND_SLICER_SEGMENTS)) {
+            BlobList parts = new BlobList();
+            File segments = new File(cmdParams.get(SEGMENTS_PATH));
+            for (File segmentFile : segments.listFiles()) {
+                FileBlob fb = new FileBlob(segmentFile);
+                fb.setFilename(segmentFile.getName());
+                fb.setMimeType(input.getBlob().getMimeType());
+                Framework.trackFile(segmentFile, parts);
+                parts.add(fb);
+            }
+            return new SimpleBlobHolder(parts);
+        }
+        Blob result = new FileBlob(new File(cmdParams.get(OUTPUT_FILE_PATH_PARAM)));
+        result.setMimeType(input.getBlob().getMimeType());
+        result.setFilename(cmdParams.get(OUTPUT_FILE_PATH_PARAM));
+        return new SimpleBlobHolder(result);
+    }
+
+    private String setupFilename(String filename, String startAt, String duration) {
+        String suffix = !StringUtils.isEmpty(startAt)? "-" + startAt.replaceAll(":", "") : "";
+        suffix += !StringUtils.isEmpty(duration)? "-" + duration.replaceAll(":", "") : "";
```

Comment:
My preference is to use the positive form always, I find it's more readable. Find and use what you prefer.